sources/packages/backend/apps/api/src/route-controllers/student/student.controller.service.ts

```diff
@@ -64,24 +74,49 @@ export class StudentControllerService {
     groupName: string,
     auditUserId: number,
   ): Promise<FileCreateAPIOutDTO> {
-    const createdFile = await this.fileService.createFile(
-      {
-        fileName: file.originalname,
-        uniqueFileName: uniqueFileName,
-        groupName: groupName,
-        mimeType: file.mimetype,
-        fileContent: file.buffer,
-      },
-      studentId,
-      auditUserId,
-    );
-
+    try {
+      this.logger.log(`Uploading file ${file.originalname} to S3 storage.`);
+      await this.objectStorageService.putObject({
+        key: uniqueFileName,
+        contentType: file.mimetype,
+        body: file.buffer,
+      });
+      this.logger.log(`Uploaded file ${file.originalname} to S3 storage.`);
+    } catch (error: unknown) {
+      this.logger.error(
+        `Uploading file ${file.originalname} to S3 storage failed. S3 storage failure.`,
+      );
+      this.logger.error(parseJSONError(error));
+      throw new InternalServerErrorException(
+        `Unexpected error while uploading the file ${file.originalname}.`,
+        FILE_UPLOAD_SERVICE_UNAVAILABLE,
+      );
+    }
```

Comment:
I think that logic for the S3 could be moved to service and dealt inside the `createFile` method as we always do it all together. Thoughts? @andrewsignori-aot @dheepak-aot @guru-aot

Reply:
Sounds good to me @andrepestana-aot.

Reply:
Updated.